snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -444,7 +441,10 @@ def prime(self, force=False):
         snap_files, snap_dirs = self.migratable_fileset_for('prime')
         _migrate_files(snap_files, snap_dirs, self.stagedir, self.primedir)
 
-        dependencies = _find_dependencies(self.primedir, snap_files)
+        elf_files = elf.get_elf_files(self.primedir, snap_files)
+        dependencies = set()
+        for elf_file in elf_files:
+            dependencies = dependencies.union(elf.get_dependencies(elf_file))
```

Comment:
This creates multiple copies of the dependencies. Would probably be more efficient to update it in place, either with `dependencies |= elf.get_dependencies(elf_file)` or `dependencies.update(elf.get_dependencies(elf_file))`.